src/sentry/static/sentry/app/components/events/interfaces/spans.tsx

```diff
@@ -0,0 +1,467 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import _ from 'lodash';
+
+import {t} from 'app/locale';
+import SentryTypes from 'app/sentryTypes';
+import GuideAnchor from 'app/components/assistant/guideAnchor';
+import EventDataSection from 'app/components/events/eventDataSection';
+
+import TraceViewMinimap from './minimap';
+
+type SpanType = {
+  trace_id: string;
+  parent_span_id?: string;
+  span_id: string;
+  start_timestamp: number;
+  timestamp: number | null | undefined; // this is essentially end_timestamp
+  same_process_as_parent: boolean;
+  op?: string;
+  data: Object;
+};
+
+type SpanEntry = {
+  type: 'spans';
+  data: SpanType[];
+};
+
+type TraceType = {
+  type: 'trace';
+  span_id: string;
+  trace_id: string;
+};
+
+const isValidSpanID = (maybeSpanID: any) => {
+  return _.isString(maybeSpanID) && maybeSpanID.length > 0;
+};
+
+type LookupType = {[span_id: string]: SpanType[]};
+
+type SpanBoundsType = {startTimestamp: number; endTimestamp: number | null | undefined};
+type SpanGeneratedBoundsType = {start: number; end: number};
+
+const boundsGenerator = (bounds: {
+  traceStartTimestamp: number;
+  traceEndTimestamp: number;
+  viewStart: number; // in [0, 1]
+  viewEnd: number; // in [0, 1]
+}) => {
+  const {traceEndTimestamp, traceStartTimestamp, viewStart, viewEnd} = bounds;
+
+  // viewStart and viewEnd are percentage values (%) of the view window relative to the left
+  // side of the trace view minimap
+
+  // invariant: viewStart <= viewEnd
+
+  // duration of the entire trace in seconds
+  const duration = traceEndTimestamp - traceStartTimestamp;
+
+  const viewStartTimestamp = traceStartTimestamp + viewStart * duration;
+  const viewEndTimestamp = traceEndTimestamp - (1 - viewEnd) * duration;
+  const viewDuration = viewEndTimestamp - viewStartTimestamp;
+
+  return (spanBounds: SpanBoundsType): SpanGeneratedBoundsType => {
+    const {startTimestamp, endTimestamp} = spanBounds;
+
+    const start = (startTimestamp - viewStartTimestamp) / viewDuration;
+
+    if (!_.isNumber(endTimestamp)) {
```

Comment:
Should you import just the functions you need from lodash?

Reply:
No longer relevant.